### PR TITLE
do not require selenium for any common imports

### LINKED ISSUE #9676: [BUG] Selenium required to build extensions
After trying to compile panel with bokeh 2.0 I encounter this problem:

https://github.com/holoviz/panel/pull/905#issuecomment-586711398

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #9676
- [x] tests added / passed

### CODE REVIEW

File: tests/codebase/test_no_selenium_common.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "import bokeh.io",
+    "import bokeh.models",
+    "import bokeh.plotting",
+    "import bokeh.server",
+]
+
+def test_no_tornado_common() -> None:
+    ''' Basic usage of Bokeh should not result in any Tornado code being
+    imported. This test ensures that importing basic modules does not bring in
+    Tornado.
```
Comment (mattpap) [inline tests/codebase/test_no_selenium_common.py]: Selenium.

File: tests/codebase/test_no_selenium_common.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2017, Anaconda, Inc. All rights reserved.
```
Comment (mattpap) [inline tests/codebase/test_no_selenium_common.py]: 2020.

File: bokeh/io/export.py [hunk 3]
```diff
@@ -203,11 +207,13 @@ def get_screenshot_as_png(obj: Union[LayoutDOM, Document], *, driver: Optional[W
 
     return Image.open(io.BytesIO(png)).convert("RGBA").crop((0, 0, width, height))
 
-def get_svgs(obj: Union[LayoutDOM, Document], *, driver: Optional[WebDriver] = None, timeout: int = 5,
+def get_svgs(obj: Union[LayoutDOM, Document], *, driver: "Optional[WebDriver]" = None, timeout: int = 5,
```
Comment (mattpap) [inline bokeh/io/export.py]: Quoting `Optional` isn't necessary, though correct.

**Comment by mattpap** [comment]
My preference would be to just not reference `io.export` in `io.__init__`. This module is optional if selenium is not available, so it shouldn't be conveniently available, or it could be conditionally available. I suppose a counter argument is that we don't want to change such high level API.

**Comment by bryevdv** [comment]
> My preference would be to just not reference io.export in io.__init__. This module is optional if selenium is not available, so it shouldn't be conveniently available, or it could be conditionally available. I suppose a counter argument is that we don't want to change such high level API.

I am open to other possibilities that keep `export_png` and `export_svgs` in `bokeh.io`. All the ones I can think of seem like much more trouble

* move those entire functions int `__init__`
* move stub functions that call the real versions into `__init__` (will have to copy docstrings over)

File: bokeh/io/export.py (hunk 3 above)
Comment (bryevdv) [reply bokeh/io/export.py]: I know, I personally prefer an "all or nothing" approach re: string types but I don't have a strong preference here so if you do feel strongly let me know.

**Comment by bryevdv** [comment]
@mattpap I'll merge this later today unless you ask me not to.

**Comment by mattpap** [comment]
This approach is fine, at least for now. If time permits, and especially if there will be more cases like this emerging, then I may purse an alternative solution, as you outlined.

**Comment by bryevdv** [comment]
FWIW I don't think there will be more cases like this. At least, I don't have many features in mind that will require optional dependencies (or selenium specifically, if that is what you mean)

**Pull request merged**